lib/shopping_cart.rb

```diff
@@ -0,0 +1,49 @@
+class ShoppingCart
+  attr_reader :name, :capacity, :products, :details, :is_full
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity.delete_suffix("items").to_i
+    @products = []
+    @details = {}
+    @details[:name] = name
+    @details[:capacity] = @capacity
+    @is_full = false
+
+  end
+
+  def add_product(product)
+    @products << product
+    total_products = products.sum do |product|
+      product.quantity
+    end
+    @is_full = true if total_products >= capacity
```

Comment:
you're missing a method, `total_number_of_products` . lines 17-19 could be in that method, and then your `is_full` method can take care of comparing the total products to the capacity (just like you're doing on 20).